src/NLog/SetupBuilderExtensions.cs

```diff
@@ -106,6 +114,13 @@ public static ISetupBuilder LoadConfiguration(this ISetupBuilder setupBuilder, A
             {
                 setupBuilder.LogFactory.Configuration = newConfig;
             }
+            if (applyOnReload)
+            {
+                setupBuilder.LogFactory.ConfigurationReloaded += (sender, e) =>
+                {
+                    setupBuilder.LoadConfiguration(false, configBuilder);
```

Comment:
Should only perform the call when `e.Succeeded == true`.

Reply:
When using `ConfigurationReloaded` instead of `ConfigurationChanged`, then it will not work together with MEL-NLogLoggingConfiguration. As it doesn't make use of internal NLog file-watchers https://github.com/NLog/NLog.Extensions.Logging/blob/4bd3bcb9aceca3da3b76dee6cbe711f0cd0fb34b/src/NLog.Extensions.Logging/Config/NLogLoggingConfiguration.cs#L111

Think you should change from `ConfigurationReloaded` to `ConfigurationChanged`. Maybe mark `ConfigurationReloaded` as obsolete in NLog 5.0?

Reply:
@304NotModified Created #3954 with example of how to use `ConfigurationChanged`